crates/storage/libmdbx-rs/src/transaction.rs

```diff
@@ -355,7 +329,7 @@ where
     K: TransactionKind,
 {
     fn drop(&mut self) {
-        self.txn_execute(|txn| {
+        let _ = self.txn_execute(|txn| {
```

Comment:
so this locks the `TransactionPtr`, and then `self.env.txn_manager().remove_active_read_transaction(txn)` removes the locked `TransactionPtr` from timed out transactions set, should work. could you please add a test for dropping a timed out txn to cover this code?

Reply:
oops, apparently it doesn't :D **any** operations on the reset transaction (including `mdbx_txn_abort`) will fail. So first, to drop the transaction, we need to renew it, and only then abort. Should be fine, given that we take a lock here and no other operations can be done on the pointer during that time.